bpf/lib/nat.h

```diff
@@ -1642,17 +1639,17 @@ static __always_inline int snat_v6_rewrite_egress(struct __ctx_buff *ctx,
 			return DROP_CSUM_L4;
 #endif  /* ENABLE_SCTP */
 		case IPPROTO_ICMPV6: {
-			__be32 from, to;
-
 			if (ctx_store_bytes(ctx, off +
 					    offsetof(struct icmp6hdr,
 						     icmp6_dataun.u_echo.identifier),
 					    &state->to_sport,
 					    sizeof(state->to_sport), 0) < 0)
 				return DROP_WRITE_ERROR;
-			from = tuple->sport;
-			to = state->to_sport;
-			sum = csum_diff(&from, 4, &to, 4, sum);
+			if (l4_csum_replace(ctx, off + offsetof(struct icmphdr, checksum),
```

Comment:
`ctx` here holds an ICMPv6 packet, maybe we should use `offsetof(struct icmp6hdr, icmp6_cksum)`?

Reply:
I should not do this at all for icmp6